File: app/src/main/java/com/medavox/diabeticdiary/CarbIngredient.java
package com.medavox.diabeticdiary;

import android.support.annotation.Nullable;

/**
 * @author Adam Howard
 * @since 01/09/2017
 */

public class CarbIngredient {

    private String name;//optional
    private int grams;
    private int percentCarb;

    public CarbIngredient(int grams, int percentCarb) {
        this.grams = grams;
        this.percentCarb = percentCarb;
    }

    public CarbIngredient(String name, int grams, int percentCarb) {
        this.name = name;
        this.grams = grams;
        this.percentCarb = percentCarb;
    }

    /**Calculate and return the total CP in this ingredient*/
    public int getCPx1000() {
        return percentCarb * grams;
    }

    @Nullable
    public String getName() {
        return name;
    }

    public int getGrams() {
        return grams;
    }

    public int getPercentCarb() {
        return percentCarb;
    }

    @Override
    public String toString() {
        return "CarbIngredient ["+(name == null ?  "" : "\""+name+"\" ")+grams+"g, "+percentCarb+"%]";
    }
}

File: app/src/main/java/com/medavox/diabeticdiary/CarbIngredient.kt
package com.medavox.diabeticdiary

/**
 * @author Adam Howard
 * @since 01/09/2017
 */
data class CarbIngredient(val grams:Int, val percentCarb:Int, val name:String?=null) {

    /**Calculate and return the total CP in this ingredient*/
    val CPx1000:Int get() {
        return percentCarb * grams
    }

    override fun toString():String {
        return "CarbIngredient ["+(if(name == null)  "" else "\"$name\" ")+grams+"g, "+percentCarb+"%]"
    }
}
